pkgs/by-name/me/memos/package.nix

```
@@ -0,0 +1,122 @@
+{
+  fetchFromGitHub,
+  buildGoModule,
+  stdenvNoCC,
+  nix-update-script,
+  pkgs,
+  lib,
+  pnpm,
+  buf,
+  cacert,
+  grpc-gateway,
+  protoc-gen-go,
+  protoc-gen-go-grpc,
+  protoc-gen-validate,
+}:
+let
+  version = "0.24.2";
+  src = fetchFromGitHub {
+    owner = "usememos";
+    repo = "memos";
+    rev = "v${version}";
+    hash = "sha256-DWOJ6+lUTbOzMLsfTDNZfhgNomajNCnNi7U1A+tqXm4=";
+  };
+
+  protobufsGenerated = stdenvNoCC.mkDerivation {
+    name = "memos-protobuf-gen";
+    inherit src;
+
+    nativeBuildInputs = [
+      buf
+      cacert
+      grpc-gateway
+      protoc-gen-go
+      protoc-gen-go-grpc
+      protoc-gen-validate
+    ];
+
+    buildPhase = ''
+      runHook preBuild
+      pushd proto
+      HOME=$TMPDIR buf generate
+      popd
+      runHook postBuild
+    '';
+    installPhase = ''
+      runHook preInstall
+      mkdir -p $out/{proto,web/src/types}
+      cp -r {.,$out}/proto/gen
+      cp -r {.,$out}/web/src/types/proto
+      runHook postInstall
+    '';
+
+    outputHashMode = "recursive";
+    outputHashAlgo = "sha256";
+    outputHash = "sha256-u+Wq/fXvWTjXdhC2h6RCsn7pjdFJ+gUdTPRvrn9cZ+k=";
+  };
+
+  frontend = stdenvNoCC.mkDerivation (finalAttrs: {
+    pname = "memos-web";
+    inherit version src;
+    pnpmDeps = pnpm.fetchDeps {
+      inherit (finalAttrs) pname version src;
+      sourceRoot = "${finalAttrs.src.name}/web";
+      hash = "sha256-lopCa7F/foZ42cAwCxE+TWAnglTZg8jY8eRWmeck/W8=";
+    };
+    pnpmRoot = "web";
+    nativeBuildInputs = with pkgs; [
```

```suggestion
    nativeBuildInputs = [
```
Overlooked this last time. Looks okay otherwise.

---

woah, i am surprised i even added that